addons/account/models/account_move.py

```diff
@@ -1565,6 +1567,8 @@ def _fetch_duplicate_supplier_reference(self, matching_states=('draft', 'posted'
                     for field_name in used_fields
                 },
             }
+            # We add this line so that the amount total is checkable before the save.
```

Comment:
I think it's not self explanatory, it just triggers more question.
Could you write that the updated amount_total isn't available in the query as long as the record hasn't been saved because it is computed from line_ids?